plugins/adobe-analytics-compatible-version/plugin.test.ts

```diff
@@ -0,0 +1,168 @@
+import {
+  ClientInfoAndroid,
+  ClientInfoIOS,
+  Configuration,
+  SharedStateVersions
+} from '@adobe/griffon-toolkit-aep-mobile';
+// @ts-ignore
+import plugin from './index';
+
+const configurationEvent: Configuration = {
+  clientId: '1',
+  timestamp: 10000,
+  type: 'client',
+  uuid: '1'
+};
+
+const versionEvent: SharedStateVersions = {
+  clientId: '1',
+  timestamp: 10000,
+  type: 'client',
+  uuid: '1',
+  payload: {
+    ACPExtensionEventData: {
+      stateowner: 'com.adobe.module.eventhub'
+    },
+    ACPExtensionEventSource: 'com.adobe.eventsource.sharedstate',
+    ACPExtensionEventType: 'com.adobe.eventtype.hub',
+    metadata: {
+      'state.data': {
+        extensions: {
+          'com.adobe.assurance': {
+            version: '3.0.0'
+          },
+          'com.adobe.module.analytics': {
+            version: '3.0.0'
+          }
+        },
+        version: '3.0.0'
+      }
+    }
+  }
+};
+
+const invalidVersionEvent: SharedStateVersions = {
+  clientId: '1',
+  timestamp: 10000,
+  type: 'client',
+  uuid: '1',
+  payload: {
+    ACPExtensionEventData: {
+      stateowner: 'com.adobe.module.eventhub'
+    },
+    ACPExtensionEventSource: 'com.adobe.eventsource.sharedstate',
+    ACPExtensionEventType: 'com.adobe.eventtype.hub',
+    metadata: {
+      'state.data': {
+        extensions: {
+          'com.adobe.assurance': {
+            version: '1.0.0'
+          },
+          'com.adobe.module.analytics': {
+            version: '1.0.0'
+          }
+        },
+        version: '1.0.0'
+      }
+    }
+  }
+};
+
+const iosClientInfo: ClientInfoIOS = {
+  clientId: '1',
+  timestamp: 10000,
+  type: 'client',
+  uuid: '1',
+  payload: {
+    type: 'connect',
+    deviceInfo: {
+      'Canonical platform name': 'iOS',
+      'Device name': 'My Device',
+      Model: 'IPhone X'
+    }
+  }
+};
+
+const iosClientInfoTwo: ClientInfoIOS = {
+  clientId: '2',
+  timestamp: 10000,
+  type: 'client',
+  uuid: '2',
+  payload: {
+    type: 'connect',
+    deviceInfo: {
+      'Canonical platform name': 'iOS',
+      'Device name': 'My Device',
+      Model: 'IPhone X'
+    }
+  }
+};
+
+const androidClientInfo: ClientInfoAndroid = {
+  clientId: '1',
+  timestamp: 10000,
+  type: 'client',
+  uuid: '1',
+  payload: {
+    deviceInfo: {
+      'Canonical platform name': 'Android'
+    }
+  }
+};
```

Comment:
Might be useful to look at the `mock` calls of event defs like https://github.com/adobe/griffon-toolkit/blob/main/packages/aep-mobile/src/sharedStateVersions.ts and others to generate mock events in less lines of code. Looks like some of them might need to be adjusted to meet these usecases